inc/commonitilobject.class.php

```diff
@@ -6815,25 +6815,36 @@ function showTimeline($rand) {
       echo "<div class='h_date'><i class='far fa-clock'></i>".Html::convDateTime($this->fields['date'])."</div>";
       echo "<div class='h_user'>";
 
-      $user->getFromDB($this->fields['users_id_recipient']);
+      $user = new User();
+      $display_requester = false;
+      $requesters = $this->getUsers(CommonITILActor::REQUESTER);
+      if (count($requesters) === 1) {
+         $requester = reset($requesters);
+         if ($requester['users_id'] > 0) {
+            // Display requester identity only if there is only one requester
+            // and only if it is not an anonymous user
+            $display_requester = $user->getFromDB($requester['users_id']);
+         }
+      }
+
       echo "<div class='tooltip_picture_border'>";
       $picture = "";
-      if (isset($user->fields['picture'])) {
+      if ($display_requester && isset($user->fields['picture'])) {
          $picture = $user->fields['picture'];
       }
       echo "<img class='user_picture' alt=\"".__s('Picture')."\" src='".
       User::getThumbnailURLForPicture($picture)."'>";
       echo "</div>";
 
-      if (isset($user->fields['id']) && $user->fields['id']) {
+      if ($display_requester) {
          echo $user->getLink()."&nbsp;";
          $reqdata = getUserName($user->getID(), 2);
          echo Html::showToolTip(
             $reqdata["comment"],
             ['link' => $reqdata['link']]
          );
       } else {
-         echo __('Requester');
+         echo _n('Requester', 'Requesters', count($requesters));
```

Comment:
Too late to change source strings for next bugfixes release

Reply:
It already exists here https://github.com/glpi-project/glpi/blob/9.4/bugfixes/inc/notificationtargetcommonitilobject.class.php#L1424